praw/objector.py

```diff
@@ -61,6 +71,41 @@ def objectify(self, data):
                 or {'id', 'name', 'permissions'}.issubset(set(data.keys()))):
             parser = self.parsers[self._reddit.config.kinds['redditor']]
             return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and (
+                {'conversation', 'messages', 'user', 'modActions'}
+                .issubset(set(data.keys()))):
+            # Modmail conversation
+            parser = self.parsers['ModmailConversation']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'actionTypeId' in data.keys():
+            # Modmail mod action
+            data = self.fix_dict_keys(data, {'actionTypeId': 'action_type'})
+            parser = self.parsers['ModmailAction']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'isInternal' in data.keys():
+            # Modmail message
+            camel_attributes = {'bodyMarkdown': 'body_markdown',
+                                'isInternal': 'is_internal'}
+            data = self.fix_dict_keys(data, camel_attributes)
+            parser = self.parsers['ModmailMessage']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'isAdmin' in data.keys():
+            # Modmail author
+            camel_attributes = {'isAdmin': 'is_admin',
+                                'isDeleted': 'is_deleted',
+                                'isHidden': 'is_hidden',
+                                'isMod': 'is_mod',
+                                'isOp': 'is_op',
+                                'isParticipant': 'is_participant'}
+            data = self.fix_dict_keys(data, camel_attributes)
+            del(data['id'])
```

Comment:
This is done because data['id'] is an `int`, which would otherwise clobber the base-36 `str` that the `Redditor` objects expect.